fathom/src/surface/elaboration.rs

```diff
@@ -1354,35 +1392,68 @@ impl<'interner, 'arena> Context<'interner, 'arena> {
 
                 (fun_type, self.universe.clone())
             }
-            Term::FunLiteral(range, patterns, body_expr) => {
-                self.synth_fun_lit(*range, patterns, body_expr, None)
+            Term::FunLiteral(range, params, body_expr) => {
+                self.synth_fun_lit(*range, params, body_expr, None)
             }
-            Term::App(range, head_expr, arg_exprs) => {
-                let head_range = head_expr.range();
+            Term::App(range, head_expr, args) => {
+                let mut head_range = head_expr.range();
                 let (mut head_expr, mut head_type) = self.synth(head_expr);
-
-                for arg_expr in *arg_exprs {
+                let mut args = args.iter().peekable();
+                while let Some(arg) = args.peek() {
                     head_type = self.elim_env().force(&head_type);
-                    match (&head_expr, head_type.as_ref()) {
-                        // Ensure that the head of the application is a function
-                        (_, Value::FunType(_, param_type, body_type)) => {
-                            // Check the argument and apply it to the body type
-                            let param_range = arg_expr.range();
-                            let arg_expr = self.check(arg_expr, param_type);
+                    match head_type.as_ref() {
+                        // plicities match: no special treatment needed
```

Comment:
Nit: some of these comments start with a lower case which is a little inconsistent with the style used in the other comments.